site/src/pages/docs/client-factory-builder.mdx

```diff
@@ -0,0 +1,134 @@
+# Customizing a `ClientFactory` with `ClientFactoryBuilder`
+
+A <type://ClientFactory> manages connections and protocol-specific properties.
+You can customize the properties via <type://ClientFactoryBuilder>. For example,
+let's say that you want to increase the connection timeout.
+
+```java
+final ClientFactory factory =
+                ClientFactory.builder()
+                             // Increase the connection timeout to 5 seconds.
+                             .connectTimeoutMillis(5000)
+                             .build());
+final WebClient client = WebClient.builder("http://armeria.com")
+                                  .factory(factory)
+                                  .build();
+client.get("/api").aggregate().join();
+```
+
+The client created with the given <type://ClientFactory> will respect the settings such as connection timeout
+that you defined.
+You might notice that the clients, which are created using the same <type://ClientFactory>,
+share the connection pool. If you don't want to share the connections among the clients,
+you should use different <typeplural://ClientFactory>.
+For the complete list of properties, see <type://ClientFactoryBuilder>.
+
+<Tip>
+
+  You might want to take a look at <type://ClientBuilder> to distinguish the client-specific properties.
+
+</Tip>
+
+You can also use <type://ClientFactoryOptions> to build <type://ClientFactory>.
+It's more verbose than calling a dedicated setter method, but it can be useful when you need to
+set a series of options programmatically.
+
+## Lifecycle of `ClientFactory`
+
+Most clients for a <type://ClientFactory> need to send requests to another server for the lifetime of a process.
+The fact is that when <type://ClientFactory> is closed for some reasons,
```

Comment:
This is too much implementation detail for end user docs, hence my previous suggestion...

Reply:
Let me remove details